### PR TITLE
{chem}[foss/2023a] LAMMPS v2Aug2023 w/ kokkos

### PR DESCRIPTION
**Pull request by laraPPr**
(created using `eb --new-pr`)^

- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/19470
- [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/19469
- [x] https://github.com/easybuilders/easybuild-easyblocks/pull/3036

### CODE REVIEW

**Comment by laraPPr** [comment]
I am also getting the error that shows up in https://github.com/easybuilders/easybuild-easyconfigs/pull/19396 
```
make[2]: *** No rule to make target 'mdi_build_ext/lib/mdi/libmdi.a', needed by 'liblammps.so.0'.       Stop.
```

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+simulation domain. The code is designed to be easy to modify or extend with new
+functionality.
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+toolchainopts = {'openmp': True, 'usempi': True}
+
+# 'https://github.com/lammps/lammps/archive/'
+source_urls = [GITHUB_LOWER_SOURCE]
+sources = ['stable_%(version)s_update2.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb:22]: In the past we haven't reflected the update in the version number, but we probably should?

PR #19396 uses `-update1-kokkos`, should we adopt something similar here?

In addition, I'm in favor of using a more sensible versioning scheme like `20230802`, but that may surprise end users who are looking for `2Aug2023`?

Thoughts on this @ocaisa ?

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb [hunk 2]
```diff
@@ -0,0 +1,167 @@
+name = 'LAMMPS'
+version = '2Aug2023'
```
Comment (boegel) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb:2]:
```suggestion
version = '2Aug2023_update2'
```

Comment (boegel) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb:2]: Note that we'll need to update `translate_lammps_version` in the LAMMPS easyblock accordingly to strip off the `_updateX` from the version (or translate it to `.X`)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb (hunk 2 above)
Comment (laraPPr) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb:2]: updated the easyblock in pr https://github.com/easybuilders/easybuild-easyblocks/pull/3036/files to accomodate add _updateX in version

**Comment by laraPPr** [comment]
Test report by @laraPPr
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
node3555.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/dd1f3288af0e03dc0f0361061a534bfe for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3555.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/03d9cf3c2f2156059d46bb3c5b4a54af for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3555.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/9c6d04790791fa6094cb4f412a171689 for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3555.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/0462fdae5ca01f5d2049b42c36cd4c3d for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3569.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/132c5691d75e79335283fd06573c2771 for a full test report.

**Comment by laraPPr** [comment]
> Test report by @laraPPr Using easyblocks from PR(s) [easybuilders/easybuild-easyblocks#3036](https://github.com/easybuilders/easybuild-easyblocks/pull/3036) **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) node3569.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3 See https://gist.github.com/laraPPr/132c5691d75e79335283fd06573c2771 for a full test report.

@boegel Can you look at what might be causing the following error `make[2]: *** No rule to make target 'mdi_build_ext/lib/mdi/libmdi.a', needed by 'liblammps.so.0'.  Stop.`

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**FAILED**
Build succeeded for 0 out of 1 (2 easyconfigs in total)
node4009.donphan.os - Linux RHEL 8.8 (Ootpa), x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 535.129.03, Python 3.11.3
See https://gist.github.com/laraPPr/601cf0e6a1cdf522ca35a459a3dc2307 for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node4009.donphan.os - Linux RHEL 8.8 (Ootpa), x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 535.129.03, Python 3.11.3
See https://gist.github.com/laraPPr/5a8fa8a7efa49856acb9f8c67574a43c for a full test report.

**Comment by laraPPr** [comment]
```
/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: error in /apps/gent/RHEL8/cascadelake-ib/software/PLUMED/2.9.0-foss-2023a/lib/libplumed.a(kernel.o)(.eh_frame); no .eh_frame_hdr table will be created

/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: liblammps.so.0: undefined reference to `fftw_mpi_local_size_many_transposed'

/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: liblammps.so.0: undefined reference to `fftw_mpi_execute_r2r'

/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: liblammps.so.0: undefined reference to `fftw_mpi_cleanup'

/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: liblammps.so.0: undefined reference to `fftw_mpi_init'

/apps/gent/RHEL8/cascadelake-ib/software/binutils/2.40-GCCcore-12.3.0/bin/      ld: liblammps.so.0: undefined reference to `fftw_mpi_plan_many_transpose'

collect2: error: ld returned 1 exit status
```

**Comment by ocaisa** [comment]
It's missing the parallel FFTW libraries: https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs/f/FFTW.MPI

**Comment by laraPPr** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Comment by ocaisa** [comment]
This is really tricky, I suspect that one of the packages is pulling in a link to `-lfftw_mpi` but that is not being passed along to the main LAMMPS build.

**Comment by laraPPr** [comment]
It might be something specific to the cluster that I am testing on.  So I'm trying on another cluster and triggered the bot to see how far they get.

**Comment by laraPPr** [comment]
On the other cluster I can find this in the log:

```
== 2024-02-02 12:05:14,357 run.py:701 WARNING Found 1 potential error (some may be harmless) in output of  make  -j 24 :
/apps/gent/RHEL8/zen2-ib/software/binutils/2.40-GCCcore-12.3.0/bin/ld: error in /apps/gent/RHEL8/zen2-ib/software/PLUMED/2.9.0-foss-2023a/lib/libplumed.a(kernel.o)(.eh_frame); no .eh_frame_hdr table will be created
```

**Comment by laraPPr** [comment]
I think I found the problem for the failing sanity checks. During the installation step only a part of the installations are done. It seem to stop abruptly somewhere in the middle compared to the log of working lammps build. 

These are the last installation references in the log 
```
-- Installing: /data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL8/zen2-ib/s     oftware/LAMMPS/2Aug2023_update2-foss-2023a-kokkos/etc/profile.d/lammps.csh

7040 -- Installing: /data/gent/vo/000/gvo00002/vsc46128/easybuild/RHEL8/zen2-ib/s     oftware/LAMMPS/2Aug2023_update2-foss-2023a-kokkos/bin/phana
```

It never gets to creating `lib/python3.9/site-packages`.

**Comment by laraPPr** [comment]
In the LAMMPS [documentation](https://docs.lammps.org/Python_install.html) they mention the following command `make install-python` but I don't know if this command has become necessary since the `2AUG` version. Or if something else is going on. 

And if this is the case for Python this might also need to be done for the other `general_packages`?

**Comment by branfosj** [comment]
From the [Stable release 2 August 2023](https://github.com/lammps/lammps/releases/tag/stable_2Aug2023) release notes:
> make install in a CMake based installation will no longer install the LAMMPS python module. make install-python can be used for that.

**Comment by laraPPr** [comment]
> From the [Stable release 2 August 2023](https://github.com/lammps/lammps/releases/tag/stable_2Aug2023) release notes:
> 
> > make install in a CMake based installation will no longer install the LAMMPS python module. make install-python can be used for that.

Thanks @branfosj

**Comment by laraPPr** [comment]
How and where do I best add this extra `install_cmd`. I tried adding it with `install_cmds` in the easyconfig but that does not seem possible because I'm getting this error in the dry-run:
```
!!!
!!! WARNING: ignoring error sequence item 1: expected str instance, list found
!!!
```

**Comment by ocaisa** [comment]
The LAMMPS easyblock already has a custom install step, I would add it there (in a version dependent way)

**Comment by laraPPr** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Comment by cgross95** [comment]
> Test report by @boegelbot Using easyblocks from PR(s) [easybuilders/easybuild-easyblocks#3036](https://github.com/easybuilders/easybuild-easyblocks/pull/3036) **FAILED** Build succeeded for 1 out of 2 (2 easyconfigs in total) cns1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/208fbeed305185a09a864b693368c4fb for a full test report.

As mentioned in #19823, my builds are failing in the same way in the sanity checks as well. It looks like it's properly installed the Python module and that's running correctly, but the `mpirun` and/or the LAMMPS examples are having problems. I'm not familiar with LAMMPS, so I'm having trouble recognizing where the issue might be. I'll do my best to dig in, and I'm happy to try out any tests on our system.

**Comment by laraPPr** [comment]
> > Test report by @boegelbot Using easyblocks from PR(s) [easybuilders/easybuild-easyblocks#3036](https://github.com/easybuilders/easybuild-easyblocks/pull/3036) **FAILED** Build succeeded for 1 out of 2 (2 easyconfigs in total) cns1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/208fbeed305185a09a864b693368c4fb for a full test report.
> 
> As mentioned in #19823, my builds are failing in the same way in the sanity checks as well. It looks like it's properly installed the Python module and that's running correctly, but the `mpirun` and/or the LAMMPS examples are having problems. I'm not familiar with LAMMPS, so I'm having trouble recognizing where the issue might be. I'll do my best to dig in, and I'm happy to try out any tests on our system.

The failing sanity checks is cause by `mpirun`. The command was added for `intel/2020a` since it wasn't causing any problems in other toolchains, see [easybuilders/easybuild-easyblocks PR #2095](https://github.com/easybuilders/easybuild-easyblocks/pull/2096). I've found the same error in `foss/2022b` and `foss/2023a` and I am checking `foss/2022a` now. I did run into it on `foss/2021b`. So older toolchains seem to remain unaffected. Once I found all the toolchains that are affected I'll implement the fix in the easyblock.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/5830dbbe1d1b165f60044e4583d8ce69 for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3551.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/01187a7621e4e6d009b58d819267ea9e for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3511.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/b1466f0639a803c460af1dbefc22f30d for a full test report.

**Comment by laraPPr** [comment]
Test report by @laraPPr
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3036
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3545.doduo.os - Linux RHEL 8.8 (Ootpa), x86_64, AMD EPYC 7552 48-Core Processor, Python 3.11.3
See https://gist.github.com/laraPPr/b05a917fd1f4bda056c05154967c9710 for a full test report.

**Review by ocaisa — DISMISSED** [review]
LGTM

**Comment by ocaisa** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Review by ocaisa — DISMISSED** [review]
LGTM

**Comment by ocaisa** [comment]
@boegelbot: please test @ jsc-zen3
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Review by ocaisa — APPROVED** [review]
LGTM

**Comment by ocaisa** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Comment by ocaisa** [comment]
@boegelbot: please test @ jsc-zen3
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Review by ocaisa — APPROVED** [review]
LGTM

**Comment by ocaisa** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3036"

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb (hunk 1 above)
Comment (ocaisa) [reply easybuild/easyconfigs/l/LAMMPS/LAMMPS-2Aug2023-foss-2023a-kokkos.eb]: I think you've got to match what LAMMPS actually uses

**Comment by ocaisa** [comment]
@boegelbot: please test @ jsc-zen3
EB_ARGS=" --include-easyblocks-from-pr=3036"

**Comment by ocaisa** [comment]
Going in, thanks @laraPPr !

**Pull request merged**